### ISSUE TITLE
Remove data set filter?

### ISSUE THREAD

**Issue by bernt-matthias** [qiime2/q2galaxy#60]
Seems that people still have problems to get the correct input for the tools: https://help.galaxyproject.org/t/re-qiime2-diversity-alpha/13332

At the moment data inputs have an option filter and a validator: https://github.com/qiime2/galaxy-tools/blob/0888694a8a6241befae7f9eed8a56bc13d4593a0/tools/suite_qiime2__diversity/qiime2__diversity__alpha.xml#L28

- The validator would tell a user if a wrong type is selected and does not allow to submit the tool. 
- The option filter just removes all non-fitting datasets, i.e. the user can not select datasets with other SemanticType

Hence, at the moment the validator is never active .. because the user can not select a wrong dataset.

Intuitively (IMO) it might be better to remove the filter and improve the message of the filter, i.e. make it more specific.

**Comment by jennaj** [comment]
Great idea about using the validator! Then maybe the filter message can point people to the new artifact viz too? I am imagining a little graph icon that goes there with a tool tip "explore artifact" or similar? https://github.com/galaxyproject/galaxy/issues/18799

**Comment by ebolyen** [comment]
I may be missing something, but I believe the reason is that it is possible to browse/select a dataset manually, in which case the option filter doesn't get used. So the Validator exists as a backstop for anyone who provides the input in the strange (non-dropdown) way. I would be interested to know if there's a way to add a more interesting message to the option filter, but I suspect it doesn't really apply, since as you mention mostly you never see a wrong dataset (unless you try very hard through the browse UI).

**Comment by bernt-matthias** [comment]
You are right and I did not know about this yet :) I tried to start a discussion https://github.com/galaxyproject/galaxy/discussions/19275

I think for the moment it might be the best to check if `no_value` validator would help, i.e. if it can be used to show a useful error message/notification for the case that there are no matching data sets in the history. 
I think both validators can have nearly the same message that point the user to the needed semantic type.

Let me just mention that the filters might render to toolform imperformant for large histories. But this is probably not a real concern here.

**Comment by ebolyen** [comment]
I'm going to close this issue, since it doesn't seem like there's anything to change necessarily (at least not for now).
